src/keymap-manager.coffee

```diff
@@ -622,10 +629,16 @@ class KeymapManager
       # event. This means the current event has removed any hope that the queued
       # key events will ever match any binding. So we will clear the state and
       # start over after replaying the events in `terminatePendingState`.
-      @terminatePendingState()
+      @terminatePendingState() if event.type is 'keyup'
```

Comment:
Because `compositionstart` happens *after* `keydown`, we need to wait until keyup to know whether or not replay should be suppressed.